c/strOrFile.c

```diff
@@ -137,28 +137,73 @@ int numCharsRead = -1;
 				findDelim[1] = '\0';
 		}
 
-		theStrOrFile->inputStrPos += strlen(str);
+		theStrOrFile->theStrPos += strlen(str);
 
 		return str;
 	}
 
 	return NULL;
-} 
+}
+
+/********************************************************************
+ sf_fputs()
+ Order of parameters matches stdio fputs().
+
+ First param is the string to append, and the second param is the
+ string-or-file container to which we wish to append.
+
+ On success, returns the number of characters written.
+ On failure, returns EOF. TODO: should we set error code?
```

Comment:
Remove TODO. Returning EOF is the error indicator for fputs, so it is sufficient for this too.